src/Collection/CollectionInterface.php

```diff
@@ -802,15 +804,15 @@ public function insert(string $path, mixed $values): CollectionInterface;
      * collection as the array keys. Keep in mind that it is valid for iterators
      * to return the same key for different elements, setting this value to false
      * can help getting all items if keys are not important in the result.
-     * @return array
+     * @return ($keepKeys is true ? array<TKey, TValue> : list<TValue>)
```

Comment:
that looks like a complex one
would it be more sensible to have those in `@phpstan-` specific ones? in case some IDEs will trip over this, including the () around it.